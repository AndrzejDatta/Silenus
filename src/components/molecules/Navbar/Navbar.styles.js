import styled from "styled-components";

export const StyledNavbar = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;
