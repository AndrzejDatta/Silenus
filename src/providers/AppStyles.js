import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
