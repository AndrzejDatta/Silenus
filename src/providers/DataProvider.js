import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
