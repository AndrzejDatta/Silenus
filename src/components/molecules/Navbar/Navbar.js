import React, { useContext } from "react";
import PropTypes from "prop-types";
import Icon from "components/atoms/icon/icon.component";
import { dataContext } from "providers/DataProvider";
import { StyledNavbar, StyledNavbarElement } from "./Navbar.styles";
const Navbar = () => {
  const { navIconsColor, changeColor } = useContext(dataContext);

  const { navIcons } = useContext(dataContext);
  const navElements = navIconsColor?.map(({ name, src, isClicked }) => {
    return (
      <StyledNavbarElement to={name === "plant" ? "/" : `/${name}`}>
        <Icon
          icon={navIcons[src]}
          size="medium"
          color={isClicked ? "green" : "black"}
          click={() => changeColor(name, src)}
        />
      </StyledNavbarElement>
    );
  });
  return <StyledNavbar>{navElements}</StyledNavbar>;
};

Navbar.propTypes = {};

export default Navbar;
//TODO: zastawnoic sie czy jak oplote to Link to bedzie spoko
