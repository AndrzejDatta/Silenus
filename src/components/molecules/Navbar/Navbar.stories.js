import React from "react";
import { storiesName } from "helpers/storiesName";
import Navbar from "./Navbar";

export default {
  title: `${storiesName.molecules}Navbar`,
  component: Navbar,
  argTypes: {},
};

const Template = (args) => <Navbar {...args} />;

export const Default = Template.bind({});

Default.args = {};
