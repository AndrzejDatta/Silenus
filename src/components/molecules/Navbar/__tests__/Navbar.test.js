import React from "react";
import Icon from "components/atoms/icon/icon.component";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen, fireEvent, within } from "@testing-library/react";
import Navbar from "../Navbar";

describe("Test Navbar Component", () => {
  it("test defaut style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <Navbar />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    screen.debug();
  });
});
