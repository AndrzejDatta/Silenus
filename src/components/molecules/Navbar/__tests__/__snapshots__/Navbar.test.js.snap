// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar Component test defaut style 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-self: center;
  position: fixed;
  width: 100vw;
  height: 5%;
  bottom: 0%;
  background-color: #fff;
}

<nav
  className="c0"
/>
`;
